### PR TITLE
Desktop: Override Terser Options [webpack]

### LINKED ISSUE #42163: [BLOCKER] Desktop Runtime Error: "no such file or directory client/server/bundler/assets-fallback.json"
### Description

Looks like a recent change to Calypso is driving a new runtime error in wp-desktop: :`no such file or directory client/server/bundler/assets-fallback.json`

As far as I can tell, what's really strange about this error is we're only getting it in CI - it seems the app can still build just fine locally.

I see in Git history there were some recent changes made in 4458c1b2d19e0dd652d615fded9a25c3561371c8 that look related, although maybe a red herring. @sgomes looks like you worked on this specific change, are you able to help take a look? Happy to pair on this as this is seems really bizarre (as it's affecting branches that were building successfully before. Maybe something in the desktop webpack config needs to be updated?).

**Context**:

We [recently added desktop end-to-end testing](https://github.com/Automattic/wp-desktop/pull/870) to the Calypso pipeline (on 5/5), but looks like there were some webpack changes merged prior to then.

### PR DESCRIPTION
**Pull request by nsakaimbo**
### Description

This PR is attempts to fix #42163.

It seems setting the package Terser config to skip Safari and mangling optimizations seems to mitigate the build time errors we've been recently experiencing for desktop. (Note: [mangling is turned off for desktop in the client config]([url](https://github.com/Automattic/wp-calypso/blob/4537033c6794449ce0a10b69d72daf5fa658d553/client/webpack.config.js#L175-L177)), too - so at the very least we should be consistent.)

TL;DR: Calypso is frequently failing to build for the desktop due to out-of-memory errors. We've isolated the breakage to the Terser plugin used for minification. Looks like the Terser plugin is [subject to quite a few out-of-memory bugs](https://slack-redir.net/link?url=https%3A%2F%2Fgithub.com%2Fwebpack-contrib%2Fterser-webpack-plugin%2Fissues%2F143), and the desktop build is especially sensitive to memory constraints. The minification step in particular seems like a significant memory hog. Bar disabling minification completely, it appears some adjustments to the Terser config are worth a try.

I've been testing [a hacked version of this fix](https://github.com/Automattic/wp-calypso/compare/master...try-terser-options) where the vars are simply removed and it's worked ~5 times in succession with the desktop build (in contrast to the config in Calypso master which is currently failing at least 90% of the time). 

However I can't seem to get the `isDesktop` switch to work via `process.env`. Not sure what I'm missing? Should I be passing the var explicitly in the `build-packages` script in package.json? (I've double-checked and the `CALYPSO_ENV` var [is being set in the build environment](https://github.com/Automattic/wp-desktop/compare/try-build-with-calypso-terser-overrides)).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -342,7 +342,7 @@
 		"stylelint": "^9.10.1",
 		"supertest": "^4.0.2",
 		"svgstore-cli": "^1.3.1",
-		"terser": "^4.6.11",
+		"terser": "4.6.13",
```
Comment (sgomes) [inline package.json:345]: You shouldn't ping this to a specific version. You can update the version number while keeping this a version range:

```suggestion
		"terser": "^4.6.13",
```

File: packages/calypso-build/webpack.config.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ const { cssNameFromFilename, shouldTranspileDependency } = require( './webpack/u
  * Internal variables
  */
 const isDevelopment = process.env.NODE_ENV !== 'production';
+const isDesktop = process.env.CALYPSO_ENV === 'desktop';
```
Comment (sgomes) [inline packages/calypso-build/webpack.config.js:30]: I don't think we want to make `calypso-build` aware of `CALYPSO_ENV`.

File: packages/calypso-build/webpack.config.js [hunk 3]
```diff
@@ -99,8 +100,8 @@ function getWebpackConfig(
 				extractComments: false,
 				terserOptions: {
 					ecma: 5,
-					safari10: true,
-					mangle: true,
+					safari10: ! isDesktop,
```
Comment (sgomes) [inline packages/calypso-build/webpack.config.js:103]: I would pass in the options from `client/webpack.config.js` instead. Note that it's already doing that for `mangle`, and you'd only have to do the same for `safari10`.

File: yarn.lock [hunk 4]
```diff
@@ -23377,7 +23377,16 @@ terser-webpack-plugin@^2.1.2, terser-webpack-plugin@^2.3.1:
     terser "^4.4.3"
     webpack-sources "^1.4.3"
 
-terser@^4.1.2, terser@^4.4.3, terser@^4.6.11, terser@^4.6.3:
+terser@4.6.13:
```
Comment (sgomes) [inline yarn.lock:23380]: This should be deduplicated. I can help you with that once you have the other changes in.

**Review by sgomes — CHANGES_REQUESTED** [review]

**Review by sirreal — CHANGES_REQUESTED** [review]
I think you'll only need to change the client webpack configuration here:

https://github.com/Automattic/wp-calypso/blob/eb895763bc969b1447d3f310ed51f1def12a7b5d/client/webpack.config.js#L169-L178

I wouldn't expect any changes in the `calypso-build` package.

File: packages/calypso-build/webpack.config.js (hunk 3 above)
Comment (sgomes) [reply packages/calypso-build/webpack.config.js:103]: Actually, on second thought, I would drop this altogether. Changing this file shouldn't affect Calypso builds at all, since Calypso has its own webpack config. And changing this in `client/webpack.config.js` will affect how the value gets calculated.

**Comment by sgomes** [comment]
> I wouldn't expect any changes in the `calypso-build` package.

The changes in this PR shouldn't even be affecting Calypso builds at all, since it doesn't use `calypso-build`'s webpack config.

It's likely the only things making a difference is the version upgrade.

**Comment by nsakaimbo** [comment]
Closing as the underlying issue seems to have been address by #42288.
